kubeflow/trainer/options/kubernetes.py

```diff
@@ -0,0 +1,268 @@
+    from kubeflow.trainer.types.types import BuiltinTrainer, CustomTrainer
+
+
+@dataclass
+class Labels(KubernetesCompatible):
+    """Add labels to the TrainJob resource metadata (.metadata.labels)."""
+
+    labels: dict[str, str]
+
+    def __call__(
+        self,
+        job_spec: dict[str, Any],
+        trainer: Optional[Union["CustomTrainer", "BuiltinTrainer"]] = None,
+    ) -> None:
+        """Apply labels to the job specification."""
+        metadata = job_spec.setdefault("metadata", {})
+        metadata["labels"] = self.labels
+
+
+@dataclass
+class Annotations(KubernetesCompatible):
+    """Add annotations to the TrainJob resource metadata (.metadata.annotations)."""
+
+    annotations: dict[str, str]
+
+    def __call__(
+        self,
+        job_spec: dict[str, Any],
+        trainer: Optional[Union["CustomTrainer", "BuiltinTrainer"]] = None,
+    ) -> None:
+        """Apply annotations to the job specification."""
+        metadata = job_spec.setdefault("metadata", {})
+        metadata["annotations"] = self.annotations
+
+
+@dataclass
+class SpecLabels(KubernetesCompatible):
+    """Add labels to derivative JobSet and Jobs (.spec.labels).
+
+    These labels will be merged with the TrainingRuntime values and applied to
+    the JobSet and Jobs created by the TrainJob.
+    """
+
+    labels: dict[str, str]
+
+    def __call__(
+        self,
+        job_spec: dict[str, Any],
+        trainer: Optional[Union["CustomTrainer", "BuiltinTrainer"]] = None,
+    ) -> None:
+        """Apply spec-level labels to the job specification."""
+        spec = job_spec.setdefault("spec", {})
+        spec["labels"] = self.labels
+
+
+@dataclass
+class SpecAnnotations(KubernetesCompatible):
+    """Add annotations to derivative JobSet and Jobs (.spec.annotations).
+
+    These annotations will be merged with the TrainingRuntime values and applied to
+    the JobSet and Jobs created by the TrainJob.
+    """
+
+    annotations: dict[str, str]
+
+    def __call__(
+        self,
+        job_spec: dict[str, Any],
+        trainer: Optional[Union["CustomTrainer", "BuiltinTrainer"]] = None,
+    ) -> None:
+        """Apply spec-level annotations to the job specification."""
+        spec = job_spec.setdefault("spec", {})
+        spec["annotations"] = self.annotations
+
+
+@dataclass
+class Name(KubernetesCompatible):
+    """Set a custom name for the TrainJob resource (.metadata.name)."""
+
+    name: str
+
+    def __call__(
+        self,
+        job_spec: dict[str, Any],
+        trainer: Optional[Union["CustomTrainer", "BuiltinTrainer"]] = None,
+    ) -> None:
+        """Apply custom name to the job specification."""
+        metadata = job_spec.setdefault("metadata", {})
+        metadata["name"] = self.name
+
+
+@dataclass
+class PodTemplateOverrides(KubernetesCompatible):
+    """Add pod template overrides to the TrainJob (.spec.podTemplateOverrides)."""
+
+    pod_template_overrides: list[PodTemplateOverride]
```

Comment:
Instead of writing:

```
options=[
        PodTemplateOverrides(pod_template_overrides=[
            PodTemplateOverride(
```

Would that be easier to write:
```
options=[
        PodTemplateOverrides(
            PodTemplateOverride(),
            PodTemplateOverride(),
```